lib/Conversion/PolyToStandard/PolyToStandard.cpp

```diff
@@ -60,11 +66,27 @@ struct ConvertAdd : public OpConversionPattern<AddOp> {
 
   using OpConversionPattern::OpConversionPattern;
 
+  // Convert add lowers a poly.add operation to arith operations. A poly.add
+  // operation is defined within the polynomial ring. Coefficients are added
+  // element-wise as elements of the ring, so they are performed modulo the
+  // coefficient modulus.
   LogicalResult matchAndRewrite(
       AddOp op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const override {
-    // TODO(https://github.com/google/heir/issues/104): implement
-    return failure();
+    auto type = adaptor.getLhs().getType();
+    auto addOp = rewriter.create<arith::AddIOp>(
+        op.getLoc(), type, adaptor.getLhs(), adaptor.getRhs());
+
+    APInt mod = op.getType().getRing().coefficientModulus();
+    assert(mod != 0 && "coefficient modulus must not be zero");
+    auto modConstOp = rewriter.create<arith::ConstantOp>(
+        addOp.getLoc(), DenseElementsAttr::get(type.cast<ShapedType>(), {mod}));
```

Comment:
> addOp.getLoc()

Nitpicking and please feel free to ignore: how about adding unrelated ops to the main "op" location. (Mostly for readability and avoid confusion)